### PR TITLE
Add ignore_plugins parameter for get_status and get_hook_caller

### PR DESCRIPTION
**Pull request by williamjamir**

### CODE REVIEW

File: hookman/hooks.py [hunk 1]
```diff
@@ -115,25 +115,38 @@ def remove_plugin(self, plugin_name: str):
...
+    def get_plugins_available(self, ignored_plugins: List[str]=None) -> Optional[List[PluginInfo]]:
         """
         Return a list of :ref:`plugin-info-api-section` that are available on ``plugins_dirs``
 
+        Optionally you can pass a list of plugins that should be ignored.
+
         The :ref:`plugin-info-api-section` is a object that holds all information related to the plugin.
         """
         plugin_config_files = hookman_utils.find_config_files(self.plugins_dirs)
-        return [PluginInfo(plugin_file, self.hooks_available) for plugin_file in plugin_config_files]
```
Comment (nicoddemus) [inline hookman/hooks.py]: You could just add `if plugin_info.name not in ignored_plugins` to this list comprehension:

```python
return [PluginInfo(x, self.hooks_available) for xin plugin_config_files 
        if x.name not in ignored_plugins]
```

Also change the signature to accept a `Sequence[str]=()`, so you don't even need to handle `None`. 😉

File: hookman/hooks.py [hunk 2]
```diff
@@ -154,25 +167,27 @@ def _bind_libs_functions_on_hook_caller(self, shared_lib_path: Path, hook_caller
...
         """
         Auxiliary methods that checks if the get_status has any conflict
         """
-        if self.get_status():
+        if self.get_status(ignored_plugins):
             raise ConflictBetweenPluginsError(
                 f"Could not get a Hook Caller due to existing conflict between installed plugins")
 
-    def get_status(self) -> List[Optional[ConflictStatus]]:
+    def get_status(self, ignored_plugins: List[str]=None) -> List[Optional[ConflictStatus]]:
```
Comment (nicoddemus) [inline hookman/hooks.py]: Ditto for the signature here: `Sequence[str]=()`.

File: hookman/hooks.py [hunk 3]
```diff
@@ -154,25 +167,27 @@ def _bind_libs_functions_on_hook_caller(self, shared_lib_path: Path, hook_caller
             cpp_func = getattr(hook_caller, hook)
             cpp_func(hooks_to_bind[hook])
 
-    def ensure_is_valid(self):
+    def ensure_is_valid(self, ignored_plugins: List[str]=None):
```
Comment (nicoddemus) [inline hookman/hooks.py]: Ditto for the signature here: `Sequence[str]=()`.

File: tests/test_hooks.py [hunk 4]
```diff
@@ -54,6 +54,20 @@ def test_get_hook_caller(simple_plugin):
...
 
 
+def test_get_hook_caller_passing_ignored_plugins(datadir, simple_plugin, simple_plugin_2):
+    plugins_dirs = [simple_plugin['path'], simple_plugin_2['path']]
+    hm = HookMan(specs=simple_plugin['specs'], plugin_dirs=plugins_dirs)
+
+    assert len(hm.get_plugins_available()) == 2
+    assert len(list((datadir / 'plugins').iterdir())) == 2
+
+    hook_caller = hm.get_hook_caller(ignored_plugins=['Simple Plugin 2'])
```
Comment (nicoddemus) [inline tests/test_hooks.py]: Hmm by caption? We should ignore by some identifier, not the caption...

File: tests/test_hooks.py [hunk 5]
```diff
@@ -132,3 +149,17 @@ def test_remove_plugin(datadir, simple_plugin, simple_plugin_2):
...
+
+
+def test_get_status(datadir, simple_plugin, simple_plugin_2):
+    plugins_dirs = [simple_plugin['path'], simple_plugin_2['path']]
+    hm = HookMan(specs=simple_plugin['specs'], plugin_dirs=plugins_dirs)
+    assert len(hm.get_plugins_available()) == 2
+
+    plugin_status = hm.get_status()
+
+    assert plugin_status
```
Comment (nicoddemus) [inline tests/test_hooks.py]: Check the actual status here please

Comment (williamjamir) [reply tests/test_hooks.py]: ✔️

File: tests/test_hooks.py (hunk 4 above)
Comment (williamjamir) [reply tests/test_hooks.py]: I will do this in another PullRequest.

Comment (williamjamir) [reply tests/test_hooks.py]: https://github.com/ESSS/hookman/issues/26

File: hookman/hooks.py [hunk 6]
```diff
@@ -115,25 +115,30 @@ def remove_plugin(self, plugin_name: str):
...
+        Optionally you can pass a list of plugins that should be ignored.
+
         The :ref:`plugin-info-api-section` is a object that holds all information related to the plugin.
         """
         plugin_config_files = hookman_utils.find_config_files(self.plugins_dirs)
-        return [PluginInfo(plugin_file, self.hooks_available) for plugin_file in plugin_config_files]
 
-    def get_hook_caller(self):
+        plugins_available = [PluginInfo(plugin_file, self.hooks_available) for plugin_file in plugin_config_files]
+        return [ plugin_info for plugin_info in plugins_available if plugin_info.name not in ignored_plugins]
```
Comment (nicoddemus) [inline hookman/hooks.py]: extra space

**Pull request merged**